packages/google_maps_flutter/google_maps_flutter/lib/src/google_map.dart

```diff
@@ -469,7 +469,6 @@ class _GoogleMapState extends State<GoogleMap> {
       this,
     );
     _controller.complete(controller);
-    unawaited(_updateTileOverlays());
```

Comment:
Not sure why this was here in the first place. Based on my testing the tile overlays work just fine without it. However this might have been the cause of this issue:
https://github.com/flutter/flutter/issues/43785#issuecomment-2492693962

That's why it's removed here.